language/tools/resource-viewer/src/lib.rs

```diff
@@ -226,6 +226,69 @@ fn pretty_print_ability_modifiers(f: &mut Formatter, abilities: AbilitySet) -> s
     Ok(())
 }
 
```

Comment:
are we sure the test covers all serialization logic below?

Reply:
It definitely does not; let me add some more cases to the E2E (or figure out if there's a more lightweight way to unit test this).